tsdb/head_append.go

```diff
@@ -34,12 +34,17 @@ import (
 // initAppender is a helper to initialize the time bounds of the head
 // upon the first sample it receives.
 type initAppender struct {
-	app  storage.Appender
-	head *Head
+	app   storage.Appender
+	head  *Head
+	hints *storage.AppendHints
 }
 
 var _ storage.GetRef = &initAppender{}
 
+func (a *initAppender) SetHints(hints *storage.AppendHints) {
+	a.hints = hints
+}
```

Comment:
Are these hints being used anywhere? Sounds like we just set them and never read them.

I think this implementation should be:

```suggestion
}

var _ storage.GetRef = &initAppender{}

func (a *initAppender) SetHints(hints *storage.AppendHints) {
	a.app.SetHints(hints)
}
```